db/view/view.cc

```diff
@@ -1278,6 +1354,12 @@ future<std::optional<utils::chunked_vector<frozen_mutation_and_schema>>> view_up
         _key = std::move(std::move(_update)->as_partition_start().key().key());
         _update_partition_tombstone = _update->as_partition_start().partition_tombstone();
         do_advance_updates = true;
+
+        if (_update_partition_tombstone) {
+            // For views that have the same partition key as base, generate an update of partition tombstone to delete
+            // the entire partition in one operation, instead of generating an update for each row.
+            generate_update(_update_partition_tombstone);
```

Comment:
I'm not thrilled that this function has the ever-so-generic name generate_update() (actually, you add two functions with the same name, and I see we have several more (!) functions with the same name - I'm not sure if it adds symmetry or adds to the confusion!).
I wonder if it even makes sense to have such a function that you only call once - you could have just inserted its code here.

Reply:
I renamed the new functions `generate_partition_tombstone_update`
I wanted to keep a similar structure to the current `generate_update` functions

Reply:
Is it true (I think I saw this, but no longer sure), you only call one of these functions once, and that function is the only one that calls the second function? If that's the case (?) maybe you don't need these functions at all and can just inline their code.

Reply:
I inlined one of the `generate_partition_tombstone_update` functions in `build_some`
now there is one `generate_partition_tombstone_update` function